designsafe/middleware.py

```diff
@@ -33,6 +34,29 @@ def process_request(self, request):
                                       'or <a href="https://www.mozilla.org/en-US/firefox/new/">Firefox</a> '
                                       'if you experience issues.')
 
+class DesignsafeProfileUpdateMiddleware:
+
+    """
+    Middleware to check if a user's profile has the update_required flag set to 
+    True, and force them to update their profile if so.
+    """
+
+    def process_request(self, request):
+        blocked_path = request.path.startswith('/data')\
```

Comment:
Prefix can also be a tuple of prefixes to lookup.
```suggestion
        blocked_path = request.path.startswith(
            ("/data", "/applications", "/rw/workspace", "/recon-portal", "/dashboard")
        )
```